packages/client/index.d.ts

```diff
@@ -9,8 +13,55 @@ export type PltClientOptions = {
   type: 'openapi' | 'graphql';
   name?: string;
   serviceId?: string;
-  getHeaders?: Function;
+  getHeaders?: () => Headers;
+}
+
+interface BuildOpenAPIClientOptions {
+  url: string;
+  path: string;
+  headers?: Headers
+}
+
+interface AbstractLogger {
+  fatal: () => void;
+  error: () => void;
+  warn: () => void;
+  info: () => void;
+  debug: () => void;
+  trace: () => void;
+}
+
+interface BuildGraphQLClientOptions {
+  url: string;
+  headers?: Headers;
+}
+
+interface Parameter {
+  in: string;
+  name: string;
+}
+
+interface MethodMetaInterface {
+  operationId?: string,
+  parameters: Parameter[]
+}
+
+interface BuildGraphQLClientOutput {
+  graphql: () => Promise<unkown>;
+  [Symbol('headers')]: Headers; 
+}
+
+interface BuildOpenAPIClientOutput {
+  [key: string]: {
+    statusCode: number;
+    headers: Headers[];
+    body: any
+  } | any
+  [Symbol('headers')]: Headers; 
 }
+export function generateOperationId(path: string, method: string, methodMeta: MethodMetaInterface): string
+export async function buildOpenAPIClient(options: BuildOpenAPIClientOptions, openTelemetry: any): Promise<BuildOpenAPIClientOutput>
```

Comment:
Minor change:
```suggestion
export async function buildOpenAPIClient(options: BuildOpenAPIClientOptions, openTelemetry: object): Promise<BuildOpenAPIClientOutput>
```

Or, looking at how is used, it seems to be a type with the following structure:
```
type OpenTelemetry = {
  startSpanClient?: (urlToCall: URL, method: string, telemetryContext: any) => object
  endSpanClient?: (span: any, res: any) => void
  setErrorInSpanClient?: (span: any, err: unknown) => void
}
```